### PR TITLE
Added custom IDE and improved other IDE launch commands

### PR DESCRIPTION
**Pull request by fahad-israr**
- Resolves https://github.com/starfixdev/starfix/issues/81
- Simplifies other ide launch methods

### CODE REVIEW

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 1]
```diff
@@ -402,17 +406,21 @@ public  void launch_editor(Path directory, String ide, String path, String fileP
...
-            else{
-                runCommand(directory.getParent(),ide,path,filePath);
+            runCommand(directory.getParent(), ide,path,filePath);
+        }
+    }
+
+    public static class CustomIDE extends IDE{
+
+        public  void launch_editor(Path directory, String ide, String path, String filePath) throws IOException, InterruptedException {
+            if(filePath.indexOf("#")>0){
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:419]: why these eclipse related checks?

Comment (fahad-israr) [reply cli/src/main/java/dev/starfix/Starfix.java:419]: That's not an eclipse related check . Its intended for URL having line numbers at the end. 

Example: https://github.com/jbangdev/jbang/blob/main/build.gradle#L12

The filePath we get will be:`jbang/blob/main/build.gradle#L12`  

Using this check we will convert it something like:
`jbang/blob/main/build.gradle:12`

I thought it'd be more meaningful (we don't know which IDE is being used but apparently this format looked more meaningful for IDEs to support) .

Comment (maxandersen) [reply cli/src/main/java/dev/starfix/Starfix.java:419]: okey - really should let each IDE subclass decide how they want to treat this.

Comment (maxandersen) [reply cli/src/main/java/dev/starfix/Starfix.java:419]: but i guess for custom IDE it can make sense.

**Pull request merged**